airbyte-integrations/bases/base-normalization/integration_tests/normalization_test_output/postgres/test_simple_streams/first_output/airbyte_incremental/scd/test_normalization/multiple_column_names_conflicts_scd.sql

```diff
@@ -0,0 +1,91 @@
+
+      
+
+  create  table "postgres".test_normalization."multiple_column_names_conflicts_scd"
+  as (
+    
+-- depends_on: ref('multiple_column_names_conflicts_stg')
+with
+
+input_data as (
+    select *
+    from "postgres"._airbyte_test_normalization."multiple_column_names_conflicts_stg"
+    -- multiple_column_names_conflicts from "postgres".test_normalization._airbyte_raw_multiple_column_names_conflicts
+),
+
+scd_data as (
+    -- SQL model to build a Type 2 Slowly Changing Dimension (SCD) table for each record identified by their primary key
+    select
+      md5(cast(coalesce(cast("id" as 
+    varchar
+), '') as 
+    varchar
+)) as _airbyte_unique_key,
+      "id",
+      "User Id",
+      user_id,
+      "User id",
+      "user id",
```

Comment:
Sorry that I don't understand how the first `User id` is not converted to lower case. The main change in the Python file is this one I think:

```diff
-            if not is_quoted and not self.needs_quotes(input_name):
-                result = input_name.lower()
+            result = input_name.lower()
```

which seems to universally convert names to lower case. Can you elaborate?

Reply:
This file you commented on is the output for Postgres, not mssql.

- In Postgres, when quoted, the identifier's case matters (so  "User Id", "User id" and "user id" are different column names)
- But as the integration test reveals and as the reported On-Call issue reports, MSSQL does not make a difference with these 3 columns names with different cases and considers them all to be equal, and thus conflicting. 

So, to mirror this behavior with MSSQL in normalization code, I universally lowercase all column identifiers for MSSQL even if they are quoted. As a consequence, normalization now detects column name conflicts and will resolve it by renaming conflicting names with extra '_1', '_2' etc

Reply:
Actually, I'm going to change the PR and isolate it to normalize this sort of casing only when looking for column name conflicts...

Reply:
I see. Thanks for the explanation.